### PR TITLE
Add "Help->Open User Data Directory" menu entry

### PR DESCRIPTION
**Pull request by pierricgimmig**
Add menu entry to navigate to Orbit's application data directory.

### CODE REVIEW

File: OrbitQt/orbitmainwindow.ui [hunk 1]
```diff
@@ -425,6 +425,7 @@ QPushButton:disabled {
     <property name="title">
      <string>Help</string>
     </property>
+    <addaction name="actionOpen_User_Data_Directory"/>
```
Comment (reichlfl) [inline OrbitQt/orbitmainwindow.ui:428]: Nit: I think we should get rid of those auto-generated names and go for a consistent style. To my understanding, the majority of our QT components are named in lower camel case (`actionServiceStackOverflow`).

File: OrbitQt/orbitmainwindow.h [hunk 2]
```diff
@@ -66,6 +66,7 @@ class OrbitMainWindow : public QMainWindow {
   void closeEvent(QCloseEvent* event) override;
 
  private slots:
+  void on_actionOpen_User_Data_Directory_triggered();
```
Comment (reichlfl) [inline OrbitQt/orbitmainwindow.h:69]: Should follow our C++ coding style, not the one suggested by Qt.

File: OrbitQt/orbitmainwindow.cpp [hunk 3]
```diff
@@ -596,6 +596,13 @@ void OrbitMainWindow::on_actionReport_Bug_triggered() {
   }
 }
 
+void OrbitMainWindow::on_actionOpen_User_Data_Directory_triggered() {
+  if (!QDesktopServices::openUrl(
+          QUrl(Path::CreateOrGetOrbitAppDataDir().string().c_str(), QUrl::StrictMode))) {
+    QMessageBox::critical(this, "Error opening URL", "Could not open Orbit directory");
```
Comment (reichlfl) [inline OrbitQt/orbitmainwindow.cpp:602]: Nit: The "URL" in the title seems quite misleading to users, I'd replace it with "directory".

**Review by reichlfl — APPROVED** [review]
Thanks for adding this. The coding style should be adjusted, but since we have inconsistencies within the same file, I'm also fine with changing it later in one go.

File: OrbitQt/orbitmainwindow.cpp [hunk 4]
```diff
@@ -596,6 +596,14 @@ void OrbitMainWindow::on_actionReport_Bug_triggered() {
   }
 }
 
+void OrbitMainWindow::on_actionOpenUserDataDirectory_triggered() {
+  if (!QDesktopServices::openUrl(
+          QUrl(Path::CreateOrGetOrbitAppDataDir().string().c_str(), QUrl::StrictMode))) {
```
Comment (beckerhe) [inline OrbitQt/orbitmainwindow.cpp:601]: I'd prefer `QString::fromStdString(Path::CreateOrGetOrbitAppDataDir().string())`

**Review by beckerhe — APPROVED** [review]
Thanks Pierric. LGTM!

**Pull request merged**